### PR TITLE
[DX-1010] Update LTS Page After Review Feedback

### PR DESCRIPTION
**Pull request by dcs3spp**
## **User description**
[DX_1010](https://tyktech.atlassian.net/browse/DX-1010)

[preview](https://deploy-preview-4051--tyk-docs.netlify.app/docs/nightly/frequently-asked-questions/long-term-support-releases)

___

## **Type**
Documentation

___

## **Description**
- The PR primarily focuses on updating the documentation for Tyk's Long-Term Support (LTS) policy.
- The section "Support Arrangements for Other Tyk Components" has been significantly revised to provide more detailed information about the support strategy for different Tyk components such as Tyk Pump, Tyk Identity Broker (TIB), MDCB, and Tyk Operator.
- The revised section explains that these components are often more standalone in nature and are subject to infrequent and minor changes compared to the core products. Therefore, aligning them with the same LTS versioning isn’t necessary or practical.
- The PR also includes minor changes in the text for better readability.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>long-term-support-releases.md</strong><dd><code>Update to the Long-Term Support (LTS) documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/frequently-asked-questions/long-term-support-releases.md
- Revised the section "Support Arrangements for Other Tyk Components" to <br>  provide more detailed information about the support strategy for <br>  different Tyk components.
- Minor changes in the text for better readability.
    </details>
  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4051/files#diff-97ca604ff0eb83bbea8156ab00f75f7889dc8d2978c49730319a2c261218020d">+10/-17</a>&nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table><hr>

<details> <summary><strong>✨ Usage guide:</strong></summary><hr> 

**Overview:**
The `describe` tool scans the PR code changes, and generates a description for the PR - title, type, summary, walkthrough and labels. The tool can be triggered [automatically](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) every time a new PR is opened, or can be invoked manually by commenting on a PR.

When commenting, to edit [configurations](https://github.com/Codium-ai/pr-agent/blob/main/pr_agent/settings/configuration.toml#L46) related to the describe tool (`pr_description` section), use the following template:
```
/describe --pr_description.some_config1=... --pr_description.some_config2=...
```
With a [configuration file](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#working-with-github-app), use the following template:
```
[pr_description]
some_config1=...
some_config2=...
```

<table><tr><td><details> <summary><strong> Enabling\disabling automation </strong></summary><hr>

- When you first install the app, the [default mode](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) for the describe tool is:
```
pr_commands = ["/describe --pr_description.add_original_user_description=true" 
                         "--pr_description.keep_original_user_title=true", ...]
```
meaning the `describe` tool will run automatically on every PR, will keep the original title, and will add the original user description above the generated description. 

- Markers are an alternative way to control the generated description, to give maximal control to the user. If you set:
```
pr_commands = ["/describe --pr_description.use_description_markers=true", ...]
```
the tool will replace every marker of the form `pr_agent:marker_name` in the PR description with the relevant content, where `marker_name` is one of the following:
  - `type`: the PR type.
  - `summary`: the PR summary.
  - `walkthrough`: the PR walkthrough.

Note that when markers are enabled, if the original PR description does not contain any markers, the tool will not alter the description at all.
        

</details></td></tr>

<tr><td><details> <summary><strong> Custom labels </strong></summary><hr>

The default labels of the `describe` tool are quite generic: [`Bug fix`, `Tests`, `Enhancement`, `Documentation`, `Other`].

If you specify [custom labels](https://github.com/Codium-ai/pr-agent/blob/main/docs/DESCRIBE.md#handle-custom-labels-from-the-repos-labels-page-gem) in the repo's labels page or via configuration file, you can get tailored labels for your use cases.
Examples for custom labels:
- `Main topic:performance` - pr_agent:The main topic of this PR is performance
- `New endpoint` - pr_agent:A new endpoint was added in this PR
- `SQL query` - pr_agent:A new SQL query was added in this PR
- `Dockerfile changes` - pr_agent:The PR contains changes in the Dockerfile
- ...

The list above is eclectic, and aims to give an idea of different possibilities. Define custom labels that are relevant for your repo and use cases.
Note that Labels are not mutually exclusive, so you can add multiple label categories.
Make sure to provide proper title, and a detailed and well-phrased description for each label, so the tool will know when to suggest it.        

</details></td></tr>

<tr><td><details> <summary><strong> Inline File Walkthrough 💎</strong></summary><hr>

For enhanced user experience, the `describe` tool can add file summaries directly to the "Files changed" tab in the PR page.
This will enable you to quickly understand the changes in each file, while reviewing the code changes (diffs).

To enable inline file summary, set `pr_description.inline_file_summary` in the configuration file, possible values are:
- `'table'`: File changes walkthrough table will be displayed on the top of the "Files changed" tab, in addition to the "Conversation" tab.
- `true`: A collapsable file comment with changes title and a changes summary for each file in the PR.
- `false` (default): File changes walkthrough will be added only to the "Conversation" tab.
<tr><td><details> <summary><strong> Utilizing extra instructions</strong></summary><hr>

The `describe` tool can be configured with extra instructions, to guide the model to a feedback tailored to the needs of your project.

Be specific, clear, and concise in the instructions. With extra instructions, you are the prompter. Notice that the general structure of the description is fixed, and cannot be changed. Extra instructions can change the content or style of each sub-section of the PR description.

Examples for extra instructions:
```
[pr_description] 
extra_instructions="""
- The PR title should be in the format: '<PR type>: <title>'
- The title should be short and concise (up to 10 words)
- ...
"""
```
Use triple quotes to write multi-line instructions. Use bullet points to make the instructions more readable.

</details></td></tr>

<tr><td><details> <summary><strong> More PR-Agent commands</strong></summary><hr> 

> To invoke the PR-Agent, add a comment using one of the following commands:  
> - **/review**: Request a review of your Pull Request.   
> - **/describe**: Update the PR title and description based on the contents of the PR.   
> - **/improve [--extended]**: Suggest code improvements. Extended mode provides a higher quality feedback.   
> - **/ask \<QUESTION\>**: Ask a question about the PR.   
> - **/update_changelog**: Update the changelog based on the PR's contents.   
> - **/add_docs** 💎: Generate docstring for new components introduced in the PR.   
> - **/generate_labels** 💎: Generate labels for the PR based on the PR's contents.   
> - **/analyze** 💎: Automatically analyzes the PR, and presents changes walkthrough for each component.   

>See the [tools guide](https://github.com/Codium-ai/pr-agent/blob/main/docs/TOOLS_GUIDE.md) for more details.
>To list the possible configuration parameters, add a **/config** comment.   
 

</details></td></tr>

</table>

See the [describe usage](https://github.com/Codium-ai/pr-agent/blob/main/docs/DESCRIBE.md) page for a comprehensive guide on using this tool.

</details>

### CODE REVIEW

File: tyk-docs/content/frequently-asked-questions/long-term-support-releases.md [hunk 1]
```diff
@@ -136,26 +136,19 @@ To receive updates on our upcoming Long-Term Support (LTS) release schedule and
...
 
-#### Enterprise Portal
+Ultimately, our diversified support approach aims to balance agility with assurance, offering the most appropriate level of support for each component while minimising risk and maximising value for our customers.
+To provide the most secure and efficient environment, we advise the following with the release of a new LTS version:
 
-We strive to avoid any long term support arrangements for our Enterprise Portal. We run a regular 6 week release cadence which delivers new capability, extension of existing capability and bug fixes. Our policy is that we aim to avoid any breaking changes, so in effect the entire Enterprise Portal is supported. Here we'd increment out version as a minor version - 1.3.0, 1.4.0, 1.5.0 etc.
+- **Developer Portal**: Continually evolves with frequent updates. We recommend adopting the latest release for optimal performance and feature set, ensuring consistent user experience and accessibility to the newest enhancements.
```
Comment (letzya) [inline tyk-docs/content/frequently-asked-questions/long-term-support-releases.md]: Should we change the bolded text of the product to headings h3 @dcs3spp

Comment (letzya) [inline tyk-docs/content/frequently-asked-questions/long-term-support-releases.md]:
```suggestion
### Developer Portal: 
Continually evolves with frequent updates. We recommend adopting the latest release for optimal performance and feature set, ensuring consistent user experience and accessibility to the newest enhancements.
```

Comment (dcs3spp) [reply tyk-docs/content/frequently-asked-questions/long-term-support-releases.md]: updated

**Comment by allaneyles** [comment]
@dcs3spp I have one final and very small change, sorry!
In the table that shows the LTS versions and their dates

We can now confidently change 'LTS+1 (version TBC)' to '5.3 LTS'

Thank you!

**Comment by dcs3spp** [comment]
> @dcs3spp I have one final and very small change, sorry! In the table that shows the LTS versions and their dates
> 
> We can now confidently change 'LTS+1 (version TBC)' to '5.3 LTS'
> 
> Thank you!

Thanks @allaneyles I have updated the release offering page to include 5.3 LTS

**Comment by allaneyles** [comment]
@dcs3spp 

can we update the version compatibility table in LTS page to the real versions:
Helm Chart 1.3.0
Operator v1.16
Tyk Sync v1.4.3
TIB v1.5.1

thank you!

**Comment by dcs3spp** [comment]
> @dcs3spp
> 
> can we update the version compatibility table in LTS page to the real versions: Helm Chart 1.3.0 Operator v1.16 Tyk Sync v1.4.3 TIB v1.5.1
> 
> thank you!

Thanks @allaneyles I've updated the compatibility table

Helm Chart 1.3.0 -> updated
Operator v1.16 -> I updated to v0.17 (recommended version) and v0.16 (backwards compatible version) from slack eng leads channel conversation
Tyk Sync v1.4.3 -> updated, _still awaiting backward compatible version(s)_. @mativm02 @sredxny can you provide this data?
TIB v1.5.1 -> updated

Preview update is available [here](https://deploy-preview-4051--tyk-docs.netlify.app/docs/nightly/frequently-asked-questions/long-term-support-releases/#compatibility)

**Comment by allaneyles** [comment]
@dcs3spp the only compatible sync version with 5.3 is sync 1.4.3

**Comment by dcs3spp** [comment]
> @dcs3spp the only compatible sync version with 5.3 is sync 1.4.3

Thanks @allaneyles I have updated. Once you have the mailing list details let me know and I'll add, then we can merge this into the original PR

**Comment by allaneyles** [comment]
@dcs3spp ok, we need to publish this now and remove the old page ! So pls can I ask:
1. That we suppress the section around the mailing list - we will need this soon, but its not ready today!
2. Publish this page into prod
3. Remove the old LTS page

Thank you

**Comment by dcs3spp** [comment]
@allaneyles I have updated to remove the mailing list section. Can you use the preview given in the PR description to check all ok?

**Comment by allaneyles** [comment]
@dcs3spp few comments on the review:
The compatibility table: we have this text The table below is an illustrative example, but shows the intent: 
Can we change that to say 'our next LTS version is 5.3. The table below shows recommended compatibility' (or something similar)
2. The section on upgrading Tyk - does the link go to a production page or a page in preview only? If it points to an existing production docs page, all good!

**Comment by dcs3spp** [comment]
> @dcs3spp few comments on the review: The compatibility table: we have this text The table below is an illustrative example, but shows the intent: Can we change that to say 'our next LTS version is 5.3. The table below shows recommended compatibility' (or something similar) 2. The section on upgrading Tyk - does the link go to a production page or a page in preview only? If it points to an existing production docs page, all good!

Thanks @allaneyles, I've updated as requested. The upgrade Tyk page links to an existing published page in production

**Review by allaneyles — APPROVED** [review]

**Comment by allaneyles** [comment]
@dcs3spp the link for the mailing list form has now arrived - so we can open up that section, and use this link:
https://pages.tyk.io/long-term-support

Thanks!

**Pull request merged**